homeassistant/components/androidtv_remote/media_player.py

```diff
@@ -0,0 +1,217 @@
+
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+    MediaType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import AndroidTVRemoteBaseEntity
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Android TV media player entity based on a config entry."""
+    api: AndroidTVRemote = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([AndroidTVRemoteMediaPlayerEntity(api, config_entry)])
+
+
+class AndroidTVRemoteMediaPlayerEntity(AndroidTVRemoteBaseEntity, MediaPlayerEntity):
+    """Android TV Remote Media Player Entity."""
+
+    _attr_assumed_state = True
+    _attr_device_class = MediaPlayerDeviceClass.TV
+    _attr_supported_features = (
+        MediaPlayerEntityFeature.PAUSE
+        | MediaPlayerEntityFeature.VOLUME_STEP
+        | MediaPlayerEntityFeature.VOLUME_MUTE
+        | MediaPlayerEntityFeature.PREVIOUS_TRACK
+        | MediaPlayerEntityFeature.NEXT_TRACK
+        | MediaPlayerEntityFeature.TURN_ON
+        | MediaPlayerEntityFeature.TURN_OFF
+        | MediaPlayerEntityFeature.PLAY
+        | MediaPlayerEntityFeature.STOP
+        | MediaPlayerEntityFeature.PLAY_MEDIA
+    )
+
+    def __init__(self, api: AndroidTVRemote, config_entry: ConfigEntry) -> None:
+        """Initialize the entity."""
+        super().__init__(api, config_entry)
+
+        self._volume_max: int | None = None
+        # These tasks are needed to create a job that sends a key press
+        # sequence that can be canceled if concurrency occurs
+        self._volume_set_task: asyncio.Task | None = None
+        self._channel_set_task: asyncio.Task | None = None
+
+    def _update_current_app(self, current_app: str) -> None:
+        """Update current app info."""
+        self._attr_app_id = current_app
+        self._attr_app_name = current_app
+
+    def _update_volume_info(self, volume_info: dict[str, str | bool]) -> None:
+        """Update volume info."""
+        if volume_info.get("max"):
+            self._volume_max = int(volume_info["max"])
+            self._attr_volume_level = int(volume_info["level"]) / self._volume_max
+            self._attr_is_volume_muted = bool(volume_info["muted"])
+            self._attr_supported_features |= MediaPlayerEntityFeature.VOLUME_SET
+        else:
+            self._volume_max = None
+            self._attr_volume_level = None
+            self._attr_is_volume_muted = None
+            self._attr_supported_features &= ~MediaPlayerEntityFeature.VOLUME_SET
```

Comment:
Supported features are not allowed to change after the start event.

Reply:
An interesting restriction. I didn't know this, thanks. I've made changes.